packages/contracts/contracts/TellerV2MarketForwarder.sol

```diff
@@ -78,29 +78,56 @@ abstract contract TellerV2MarketForwarder is Initializable, ContextUpgradeable {
     ) internal virtual returns (uint256 bidId) {
         bytes memory responseData;
 
-        bytes4 submitBidSelector = bytes4(
-            keccak256(
-                "ITellerV2.submitBid(address,uint256,uint256,uint32,uint16,string,address)"
-            )
+        responseData = _forwardCall(
+            abi.encodeWithSelector(
+                //ITellerV2.submitBid.selector,
+                bytes4(keccak256("submitBid(address,uint256,uint256,uint32,uint16,string,address)")),
+                _createLoanArgs.lendingToken,
+                _createLoanArgs.marketId,
+                _createLoanArgs.principal,
+                _createLoanArgs.duration,
+                _createLoanArgs.interestRate,
+                _createLoanArgs.metadataURI,
+                _createLoanArgs.recipient
+            ),
+            _borrower
         );
 
+        return abi.decode(responseData, (uint256));
+    }
+
+  /**
+     * @notice Creates a new loan using the TellerV2 lending protocol.
+     * @param _createLoanArgs Details describing the loan agreement.]
+     * @param _borrower The borrower address for the new loan.
+     */
+    function _submitBidWithCollateral(
+        CreateLoanArgs memory _createLoanArgs,
+        Collateral[] calldata _collateralInfo,
+        address _borrower
+    ) internal virtual returns (uint256 bidId) {
+        bytes memory responseData;
+
         responseData = _forwardCall(
             abi.encodeWithSelector(
-                submitBidSelector,
+                //ITellerV2.submitBid.selector,
+                bytes4(keccak256("submitBid(address,uint256,uint256,uint32,uint16,string,address,Collateral[])")),
```

Comment:
does this work? I would imagine the encoder wouldn't know about `Collateral` struct and we would have to specify the internal values of the struct as a tuple 
i.e. instead of:
```
Collateral[]
```
it would be
```
(uint8,uint256,uint256,address)[]
```
(`CollateralType` represents `uint8`)
![image](https://user-images.githubusercontent.com/6958300/214329127-ab4210f9-6a94-47ef-b113-67addba2b71c.png)